File: ForexConnectAPI-1.1.2-Linux-i686/samples/java/SimpleSamples/TableManagerSamples/SearchInTable/src/TableListener.java
package searchintable;

import com.fxcore2.IO2GTableListener;
import com.fxcore2.O2GRow;
import com.fxcore2.O2GTableStatus;

public class TableListener implements IO2GTableListener {
    
    // Implementation of IO2GTableListener interface public method onAdded
     public void onAdded(String rowID, O2GRow rowData) {   
    }

    // Implementation of IO2GTableListener interface public method onChanged
    public void onChanged(String rowID, O2GRow rowData) {
    }
    
    // Implementation of IO2GTableListener interface public method onDeleted
    public void onDeleted(String rowID, O2GRow rowData) {
    }

    // Implementation of IO2GTableListener interface public method onStatus
    public void onStatusChanged(O2GTableStatus status) {
        switch(status) {
            case REFRESHED:
                System.out.println("\nYour refresh request was successfull");
                break;
            case REFRESHING:
            case INITIAL:
            case FAILED:
               break;
        }
    }
}

File: ForexConnectAPI-1.1.2-Linux-i686/samples/cpp/SimpleSamples/TableManagerSamples/CreateOCO/source/tablelistener.cpp
#include "tablelistener.h"

TableListener::TableListener(void)
{
    mEvent = uni::CreateEvent(0, FALSE, FALSE, 0);
    mRef = 1;
}

TableListener::~TableListener(void)
{
    uni::CloseHandle(mEvent);
}

long TableListener::addRef()
{
    return InterlockedIncrement(&mRef);
}

long TableListener::release()
{
    InterlockedDecrement(&mRef);
    if(mRef == 0)
        delete this;
    return mRef;
}

HANDLE TableListener::getEvent()
{
    return mEvent;
}

void TableListener::onStatusChanged(O2GTableStatus status)
{
    switch(status)
    {
    case Refreshed:
        uni::SetEvent(mEvent);
        break;
    case Refreshing:
    case Initial:
    case Failed:
        break;
    }
}

void TableListener::onAdded(const char *rowID, IO2GRow *row)
{
    IO2GOrderRow *orderRow = static_cast<IO2GOrderRow *>(row);
    for (unsigned int i = 0; i < mVect.size(); ++i)
    {
        if( mVect[i] == orderRow->getRequestID() )
        {
            printf("added order. id: %s\n",orderRow->getOrderID());
            mReqsComplete++;
        }
    }
    if( mReqsComplete == mVect.size() )
    {
        uni::SetEvent(mEvent);
        return;
    }
}

void TableListener::onChanged(const char *rowID, IO2GRow *row)
{
    IO2GOrderRow *orderRow = static_cast<IO2GOrderRow *>(row);
    for (unsigned int i = 0; i < mVect.size(); ++i)
    {
        if( mVect[i] == orderRow->getRequestID() )
        {
            printf("changed order. id: %s\n",orderRow->getOrderID());
            mReqsComplete++;
        }
    }
    if( mReqsComplete == mVect.size() )
    {
        uni::SetEvent(mEvent);
        return;
    }
}

void TableListener::onDeleted(const char *rowID, IO2GRow *row)
{
}

